src/components/sbb-dialog/sbb-dialog.tsx

```diff
@@ -254,12 +253,21 @@ export class SbbDialog implements ComponentInterface {
 
   // Check if the pointerdown event target is triggered on the dialog.
   private _pointerDownListener = (event: PointerEvent): void => {
-    this._isPointerDownEventOnDialog = isEventOnElement(this._dialog, event);
+    this._isPointerDownEventOnDialog = event
+      .composedPath()
+      .filter((e) => e instanceof window.HTMLElement)
+      .some((target) => (target as HTMLElement).tagName === 'DIALOG');
```

Comment:
Not sure if we maybe should give the dialog an id to be more refactoring stable? But however, some automatic tests would help to keep this improvement stable, can you create 1 to 2 e2e tests?